### PR TITLE
[Leia] cleanups, update libjpeg-turbo, addon description increase, icon add

### PR DESCRIPTION
**Pull request by AlwinEsch**
Bildschirmfoto von 2020-02-07 14-42-15

### CODE REVIEW

File: depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
++++ b/CMakeLists.txt
+@@ -40,7 +40,11 @@
+ 
+ # Detect CPU type and whether we're building 64-bit or 32-bit code
+ math(EXPR BITS "${CMAKE_SIZEOF_VOID_P} * 8")
+-string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
++if(CMAKE_SYSTEM_PROCESSOR)
++  string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
++elseif(CPU)
++  string(TOLOWER ${CPU} CMAKE_SYSTEM_PROCESSOR_LC)
```
Comment (Rechi) [inline depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: No hacks, set CMAKE_SYSTEM_PROCESSOR in the toolchain.

**Review by Rechi — CHANGES_REQUESTED** [review]

File: depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch (hunk 1 above)
Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: I look about a fix inside Kodi, it was not reproducible at home, thats why I added this hack. 

Also for things like them, I add you as reviewer, but please write with a bit nicer words :pensive:!

Comment (Rechi) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: For which platform did you try to reproduce?

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: Has tested on Linux and my Mac, for a unknown reason was there the `CMAKE_SYSTEM_PROCESSOR` available but by Jenkins not.

Here a old build about, without them https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Fimagedecoder.mpo/detail/PR-8/2/pipeline

Comment (Rechi) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: You are building the add-on the wrong way, for OSX?

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: I start today from here https://github.com/xbmc/xbmc/blob/master/tools/depends/xbmc-addons.include to find the reason why this not becomes given.

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: On OSX I use normally the small step like on Linux, but also for cases like them the same way like Jenkins over the make.

```sh
make -j $BUILDTHREADS ADDONS=imagedecoder.mpo ADDONS_DEFINITION_DIR=`pwd`/addons ADDON_SRC_PREFIX=`pwd` EXTRA_CMAKE_ARGS="-DPACKAGE_ZIP=ON -DPACKAGE_DIR=`pwd`/../../../../cmake/addons/build/zips" PACKAGE=1
```

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: Have created here a request about: https://github.com/xbmc/xbmc/pull/17363

To become usable for here then also a backport to Leia is needed.

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: @Rechi would relate it to Leia, wouldn't it be better to do it here with the Hacky Patch? And at Matrix a revised build system and omit this hack?

To make this suitable for Leia on Kodi would be a little much, with things already released.

Comment (Rechi) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: No hacks. If you can't fix it the right way, just keep the current library version, which works fine.

Comment (AlwinEsch) [reply depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch]: I look then also for backward to fix this https://github.com/xbmc/imagedecoder.raw/issues/20
For here OK to take the old but the https://github.com/xbmc/imagedecoder.raw/pull/22 contains the same.

**Review by Rechi — CHANGES_REQUESTED** [review]
Using precompiled libraries is wrong.

**Comment by AlwinEsch** [comment]
On other addons you has allowed binaries for Windows UWP, also use Kodi itself them.

Please, please take a look that you are editing this system so that this CMAKE_SYSTEM_PROCESSOR is finally available everywhere.

**Comment by AlwinEsch** [comment]
Because of this, https://github.com/xbmc/xbmc/pull/17416#issuecomment-596995617 had already referred to it here.

With:
At the moment this is only necessary with libjpeg-turbo, all other dependencies used have a fallback of this value. Since libjpeg-turbo are now in our mirror as binary I could use the libjpeg-turbo-2.0.3-*-v141-20200105.7z files for the UWP.

File: imagedecoder.mpo/addon.xml.in [hunk 2]
```diff
@@ -11,8 +11,13 @@
     extension=".mpo"
     mimetype="image/mpo"/>
   <extension point="xbmc.addon.metadata">
-    <summary>libmpo based image decoder</summary>
-    <description>libmpo based image decoder</description>
+    <summary>Multi Picture Object (MPO) image decoder</summary>
+    <description>A addon to decode MPO (multiple picture object) files. These are used mainly for 3D cameras and the Nintendo 3DS.</description>
```
Comment (garbear) [inline imagedecoder.mpo/addon.xml.in]: "A addon" -> "An addon"

Comment (AlwinEsch) [reply imagedecoder.mpo/addon.xml.in]: Thanks, have changed to them!

**Pull request merged**